src/main/java/org/example/springboot/exception/CustomGlobalExceptionHandler.java

```diff
@@ -31,4 +31,9 @@ public ResponseEntity<String> handleEntityNotFoundException(DataProcessingExcept
     public ResponseEntity<String> handleEntityNotFoundException(EntityNotFoundException ex) {
         return new ResponseEntity<>("Entity not found exception occurred", HttpStatus.NOT_FOUND);
     }
+
+    @ExceptionHandler(RegistrationException.class)
+    public ResponseEntity<String> handleRegistrationException(RegistrationException ex) {
+        return new ResponseEntity<>("Registration exception occurred", HttpStatus.NOT_FOUND);
```

Comment:
HttpStatus.NOT_FOUND implies that some data is missing on backend. This looks more like HttpStatus.BAD_REQUEST